pkg/asset/installconfig/kubevirt/client.go

```diff
@@ -0,0 +1,255 @@
+		return nil, err
+	}
+	return result, nil
+}
+
+// VirtualMachine
+
+func (c *client) GetVirtualMachine(namespace string, name string) (*kubevirtapiv1.VirtualMachine, error) {
+	resp, err := c.getResource(namespace, name, vmRes())
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, err
+		}
+		msg := fmt.Sprintf("Failed to get VirtualMachine, with error: %v", err)
+		log.Printf("[Error] %s", msg)
+		return nil, fmt.Errorf(msg)
+	}
+	var vm kubevirtapiv1.VirtualMachine
+	err = c.fromUnstructedToInterface(*resp, &vm, "VirtualMachine")
+	return &vm, err
+}
+
+func (c *client) ListVirtualMachine(namespace string) (*kubevirtapiv1.VirtualMachineList, error) {
+	resp, err := c.listResource(namespace, vmRes())
+	if err != nil {
+		msg := fmt.Sprintf("Failed to list VirtualMachine, with error: %v", err)
+		log.Printf("[Error] %s", msg)
+		return nil, fmt.Errorf(msg)
+	}
+	var vmList kubevirtapiv1.VirtualMachineList
+	err = c.fromUnstructedListToInterface(*resp, &vmList, "VirtualMachineList")
+	return &vmList, err
+}
+
+func (c *client) DeleteVirtualMachine(namespace string, name string) error {
+	return c.deleteResource(namespace, name, vmRes())
+}
+
+func vmRes() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    kubevirtapiv1.GroupVersion.Group,
+		Version:  kubevirtapiv1.GroupVersion.Version,
+		Resource: "virtualmachines",
+	}
+
+}
+
+// DataVolume
+
+func (c *client) GetDataVolume(namespace string, name string) (*cdiv1.DataVolume, error) {
+	resp, err := c.getResource(namespace, name, dvRes())
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, err
+		}
+		msg := fmt.Sprintf("Failed to get DataVolume, with error: %v", err)
+		log.Printf("[Error] %s", msg)
+		return nil, fmt.Errorf(msg)
+	}
+	var dv cdiv1.DataVolume
+	err = c.fromUnstructedToInterface(*resp, &dv, "DataVolume")
+	return &dv, nil
+}
+
+func (c *client) ListDataVolume(namespace string) (*cdiv1.DataVolumeList, error) {
+	resp, err := c.listResource(namespace, dvRes())
+	if err != nil {
+		msg := fmt.Sprintf("Failed to list DataVolume, with error: %v", err)
+		log.Printf("[Error] %s", msg)
+		return nil, fmt.Errorf(msg)
+	}
+	var dvList cdiv1.DataVolumeList
+	err = c.fromUnstructedListToInterface(*resp, &dvList, "DataVolumeList")
+	return &dvList, err
+}
+
+func (c *client) DeleteDataVolume(namespace string, name string) error {
+	return c.deleteResource(namespace, name, dvRes())
+}
+
+func dvRes() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    cdiv1.SchemeGroupVersion.Group,
+		Version:  cdiv1.SchemeGroupVersion.Version,
+		Resource: "datavolumes",
+	}
+}
+
+// Secret
+
+func (c *client) GetSecret(namespace string, name string) (*corev1.Secret, error) {
+	return c.kubernetesClient.CoreV1().Secrets(namespace).Get(context.Background(), name, metav1.GetOptions{})
+}
+
+func (c *client) ListSecret(namespace string) (*corev1.SecretList, error) {
```

Comment:
add context?

Reply:
The context is created internally in the client using `context.Background()`